### PR TITLE
client: add client option descriptions

### PR DESCRIPTION
**Pull request by batrick**
Fixes: http://tracker.ceph.com/issues/22933

Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

### CODE REVIEW

File: src/common/options.cc [hunk 1]
```diff
@@ -6942,102 +6942,102 @@ std::vector<Option> get_mds_options() {
...
     Option("client_mount_timeout", Option::TYPE_FLOAT, Option::LEVEL_ADVANCED)
     .set_default(300.0)
-    .set_description(""),
+    .set_description("timeout for client mount of CephFS"),
 
-    Option("client_tick_interval", Option::TYPE_FLOAT, Option::LEVEL_ADVANCED)
+    Option("client_tick_interval", Option::TYPE_FLOAT, Option::LEVEL_DEV)
     .set_default(1.0)
-    .set_description(""),
+    .set_description("seconds between Client upkeep ticks"),
```
Comment (smithfarm) [inline src/common/options.cc]: why is Client written with initial cap here?

File: src/common/options.cc [hunk 2]
```diff
@@ -6942,102 +6942,102 @@ std::vector<Option> get_mds_options() {
...
 
-    Option("client_trace", Option::TYPE_STR, Option::LEVEL_ADVANCED)
+    Option("client_trace", Option::TYPE_STR, Option::LEVEL_DEV)
     .set_default("")
-    .set_description(""),
+    .set_description("file containing trace of Client operations"),
```
Comment (smithfarm) [inline src/common/options.cc]: the Client with initial capital C doesn't look right

File: src/common/options.cc [hunk 3]
```diff
@@ -6942,102 +6942,102 @@ std::vector<Option> get_mds_options() {
...
 
     Option("client_reconnect_stale", Option::TYPE_BOOL, Option::LEVEL_ADVANCED)
     .set_default(false)
-    .set_description(""),
+    .set_description("reconnect when the session becomes stale"),
 
     Option("client_snapdir", Option::TYPE_STR, Option::LEVEL_ADVANCED)
     .set_default(".snap")
-    .set_description(""),
+    .set_description("pseudo directory for snapshot access of a directory"),
```
Comment (smithfarm) [inline src/common/options.cc]: "access to" a directory sounds better than "access of"

File: src/common/options.cc [hunk 4]
```diff
@@ -7065,19 +7065,20 @@ std::vector<Option> get_mds_client_options() {
...
 
     Option("client_acl_type", Option::TYPE_STR, Option::LEVEL_ADVANCED)
     .set_default("")
-    .set_description(""),
+    .set_description("ACL type to enforce (none or \"posix_acl\")"),
 
     Option("client_permissions", Option::TYPE_BOOL, Option::LEVEL_ADVANCED)
     .set_default(true)
-    .set_description(""),
+    .set_description("client enforced permission checking"),
```
Comment (smithfarm) [inline src/common/options.cc]: client-enforced with a hyphen?

File: src/common/options.cc [hunk 5]
```diff
@@ -7086,49 +7087,50 @@ std::vector<Option> get_mds_client_options() {
...
 
     Option("fuse_require_active_mds", Option::TYPE_BOOL, Option::LEVEL_ADVANCED)
     .set_default(true)
-    .set_description(""),
+    .set_description("require active MDSs in the file system when mounting"),
 
     Option("fuse_syncfs_on_mksnap", Option::TYPE_BOOL, Option::LEVEL_ADVANCED)
     .set_default(true)
-    .set_description(""),
+    .set_description("synchornize all local metadata/file chaanges after snapshot"),
```
Comment (smithfarm) [inline src/common/options.cc]: synchronize (sp)

**Comment by batrick** [comment]
@smithfarm thanks, fixed!

File: src/common/options.cc [hunk 6]
```diff
@@ -7086,49 +7086,50 @@ std::vector<Option> get_mds_client_options() {
...
 
     Option("fuse_require_active_mds", Option::TYPE_BOOL, Option::LEVEL_ADVANCED)
     .set_default(true)
-    .set_description(""),
+    .set_description("require active MDSs in the file system when mounting"),
 
     Option("fuse_syncfs_on_mksnap", Option::TYPE_BOOL, Option::LEVEL_ADVANCED)
     .set_default(true)
-    .set_description(""),
+    .set_description("synchronize all local metadata/file chaanges after snapshot"),
```
Comment (ukernel) [inline src/common/options.cc]: chaange

File: src/common/options.cc [hunk 7]
```diff
@@ -7051,7 +7050,7 @@ std::vector<Option> get_mds_client_options() {
     .set_default(0)
     .set_description(""),
 
-    Option("client_max_inline_size", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
+    Option("client_max_inline_size", Option::TYPE_UINT, Option::LEVEL_DEV)
     .set_default(4_K)
     .set_description(""),
```
Comment (jtlayton) [inline src/common/options.cc]: No description?

File: src/common/options.cc [hunk 8]
```diff
@@ -6942,102 +6942,101 @@ std::vector<Option> get_mds_options() {
...
-    Option("client_use_random_mds", Option::TYPE_BOOL, Option::LEVEL_ADVANCED)
+    Option("client_use_random_mds", Option::TYPE_BOOL, Option::LEVEL_DEV)
     .set_default(false)
-    .set_description(""),
+    .set_description("issue new requests to a random active MDS"),
 
     Option("client_mount_timeout", Option::TYPE_FLOAT, Option::LEVEL_ADVANCED)
     .set_default(300.0)
-    .set_description(""),
+    .set_description("timeout for client mount of CephFS"),
```
[the remark this answers is not in the document]
Comment (jtlayton) [reply src/common/options.cc]: Should also mention the units here (seconds).

File: src/common/options.cc [hunk 9]
```diff
@@ -6942,102 +6942,101 @@ std::vector<Option> get_mds_options() {
...
 
     Option("client_oc_target_dirty", Option::TYPE_INT, Option::LEVEL_ADVANCED)
     .set_default(8_M)
-    .set_description(""),
+    .set_description("target size of dirty pages object cache"),
 
     Option("client_oc_max_dirty_age", Option::TYPE_FLOAT, Option::LEVEL_ADVANCED)
     .set_default(5.0)
-    .set_description(""),
+    .set_description("maximum age of dirty pages in object cache"),
```
Comment (jtlayton) [inline src/common/options.cc]: "maximum age of dirty pages in object cache in seconds"

**Comment by batrick** [comment]
@smithfarm @jtlayton @ukernel please review again.

**Comment by ukernel** [comment]
LGTM

**Pull request merged**